src/app/shared/services/route.service.ts

```diff
@@ -8,24 +8,46 @@ import {
   RouterStateSnapshot,
 } from '@angular/router';
 
-import { Observable } from 'rxjs';
-import { select, Store } from '@ngrx/store';
+import { combineLatest, Observable } from 'rxjs';
+import { createSelector, MemoizedSelector, select, Store } from '@ngrx/store';
 import { isEqual } from 'lodash';
 
-import { AppState } from '../../app.reducer';
 import { AddUrlToHistoryAction } from '../history/history.actions';
 import { historySelector } from '../history/selectors';
+import { SetParametersAction, SetQueryParametersAction } from './route.action';
+import { CoreState } from '../../core/core.reducers';
+import { hasValue } from '../empty.util';
+import { coreSelector } from '../../core/core.selectors';
+
+export const routeParametersSelector = createSelector(
+  coreSelector,
+  (state: CoreState) => state.route.params
+);
+export const queryParametersSelector = createSelector(
+  coreSelector,
+  (state: CoreState) => state.route.queryParams
+);
+
+export const routeParameterSelector = (key: string) => parameterSelector(key, routeParametersSelector);
+export const queryParameterSelector = (key: string) => parameterSelector(key, queryParametersSelector);
+
+export function parameterSelector(key: string, paramsSelector: (state: CoreState) => Params): MemoizedSelector<CoreState, string> {
```

Comment:
could you add comments in this function?

Reply:
Added TypeDoc for all selectors